src/Middleware/RequestThrottling/test/RequestQueueTests.cs

```diff
@@ -12,7 +12,7 @@ public class RequestQueueTests
         [Fact]
         public async Task LimitsIncomingRequests()
         {
-            using var s = new RequestQueue(1);
+            using var s = new RequestQueue(maxConcurrentRequests: 1, requestQueueLimit: 999);
```

Comment:
Let's add a second constructor that only has a parameter for maxConcurrentRequests and default requestQueueLimit to 5000. Then have some pretty constructor chaining.

Reply:
Meh, then you're defining the default in multiple places.

Reply:
Maybe we can make the default an internal static int within options?